src/library_formatString.js

```diff
@@ -8,7 +8,7 @@ mergeInto(LibraryManager.library, {
   //   format: A pointer to the format string.
   //   varargs: A pointer to the start of the arguments list.
   // Returns the resulting string string as a character array.
-  _formatString__deps: ['strlen', '_reallyNegative'],
+  _formatString__deps: ['_reallyNegative'],
```

Comment:
We had the rule that if JS library function depends on a C function, that dependency should live in `src/deps_info.json` instead of here, and `_formatString` is a JS function and `strlen` is a C function, so `_formatString` dependency to `strlen` should be in deps_info.json and not here.

So this change would be correct under that rule. I also tried to remove it here in the earlier PR, but that caused wasm backend to fail linking. (for fastcomp backend it worked out).

I think to work around some kind of linking bug in wasm backend leads you to add the `strlen` dependency incorrectly to `emscripten_log` instead of `_formatString`?

Reply:
See the above comment, but to add to the details here: I think wasm backend linking is working as expected. My guess is that something in fastcomp happens to link in `strlen` for another reason anyhow, so the problem isn't seen (but I didn't investigate the fastcomp side).

Reply:
Hmm, the explanation above does make sense. It is unfortunate we cannot add any comments to `deps_info.json`, that is very unintuitive. LGTM then!

Reply:
Hmm, the explanation above does make sense. It is unfortunate we cannot add any comments to `deps_info.json`, that is very unintuitive. LGTM then!

Reply:
Yeah, the entire deps_info mechanism is quite confusing. Luckily it's needed rarely, and we will move more code from JS into C over time, so it will decrease. To fully get rid of it we'd need to make lld aware of the JS linking side, though, which is hard, so I imagine something like deps_info will remain in some form.